WordPress/Classes/ViewRelated/NUX/SignupGoogleViewController.swift

```diff
@@ -25,80 +25,129 @@ class SignupGoogleViewController: LoginViewController {
+    ///
+    func createWordPressComUser(googleUser: GIDGoogleUser, googleToken: String, googleEmail: String) {
         SVProgressHUD.show(withStatus: NSLocalizedString("Completing Signup", comment: "Shown while the app waits for the site creation process to complete."))
 
-        let context = ContextManager.sharedInstance().mainContext
-        let service = SignupService(managedObjectContext: context)
-        let credentials = WordPressCredentials.wpcom(username: email, authToken: token, isJetpackLogin: isJetpackLogin, multifactor: false)
+        let service = SignupService()
 
-        service.createWPComUserWithGoogle(token: token, success: { [weak self] (accountCreated) in
-            SVProgressHUD.dismiss()
-            if accountCreated {
-                self?.showSignupEpilogue(for: credentials)
-                WordPressAuthenticator.track(.signupSocialSuccess)
-            } else {
-                self?.showLoginEpilogue(for: credentials)
-                WordPressAuthenticator.track(.loginSocialSuccess)
+        service.createWPComUser(googleToken: googleToken, success: { [weak self] accountCreated, wpcomUsername, wpcomToken in
+
+            let credentials = WordPressCredentials.wpcom(username: wpcomUsername, authToken: wpcomToken, isJetpackLogin: false, multifactor: false)
+            self?.authenticationDelegate.sync(credentials: credentials) { _ in
+                if accountCreated {
+                    self?.socialSignupWasSuccessful(with: credentials)
+                } else {
+                    self?.socialLoginWasSuccessful(with: credentials)
+                }
             }
-        }) { [weak self] (error) in
+
+        }, failure: { [weak self] error in
             SVProgressHUD.dismiss()
-            WordPressAuthenticator.track(.signupSocialFailure)
-            guard let error = error else {
-                self?.navigationController?.popViewController(animated: true)
-                return
-            }
-            self?.titleLabel?.textColor = WPStyleGuide.errorRed()
-            self?.titleLabel?.text = NSLocalizedString("Google sign up failed.",
-                                                       comment: "Message shown on screen after the Google sign up process failed.")
-            self?.displayError(error as NSError, sourceTag: .wpComSignup)
-        }
+            self?.socialSignupDidFail(with: error)
+        })
+    }
+
+
+    /// Social Signup Successful: Analytics + Pushing the Signup Epilogue.
+    ///
+    func socialSignupWasSuccessful(with credentials: WordPressCredentials) {
+        WordPressAuthenticator.track(.createdAccount, properties: ["source": "google"])
+        WordPressAuthenticator.track(.signupSocialSuccess)
+
+        showSignupEpilogue(for: credentials)
+    }
+
+    /// Social Login Successful: Analytics + Pushing the Login Epilogue.
+    ///
+    func socialLoginWasSuccessful(with credentials: WordPressCredentials) {
```

Comment:
`socialLoginWasSuccessful` is a bit misleading, because we didn't expect to login. Something like `wasLoggedInInstead` explains what happened a bit better (tho it's an attrocious method name) 😜 

Also we don't _know_ if the call was successful. We just know it's complete 😉

Reply:
Sounds reasonable, updating!